src/ocean/util/app/DaemonApp.d

```diff
@@ -578,7 +578,7 @@ public abstract class DaemonApp : Application,
                     this.epoll.shutdown();
                 break;
             default:
-                assert(false);
```

Comment:
Oof.

Reply:
Given the SIGTERM logic, Is the comment above still correct?

>See ISignalExtExtension documentation for more information on how to override this method with own behaviour. `super.onSignal` is not needed to be called when doing so.

Reply:
Yes, it's still correct - if you're aren't `SIGTERM` in your app (like dhtnode) you still don't need to call this method. If you are not, this will still do what you probably want - terminate application (by shutting epoll) on SIGTERM. It's just the assert that was wrong, as the SignalExt calls all registered clients with all signals (no matter if you don't ask for that signal).

Reply:
There's at least one negative too many in your comment ;)

My point is that the derived class would have to reimplement the SIGTERM handling behaviour relating to the scheduler. Is that desirable?

Reply:
(If it doesn't call `super.onSignal`, that is.)

Reply:
> Is that desirable?

Yes. Assumption is that app knows better how to cleanly terminate itself - this is only default implementation that is expected to be not worse than crashing on unhandled signal.

Reply:
I see. Fair enough.